floor/writer.go

```diff
@@ -179,16 +184,32 @@ func (m *reflectMarshaller) decodeValue(field interfaces.MarshalElement, value r
 	case reflect.Bool:
 		field.SetBool(value.Bool())
 		return nil
-	case reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32:
+	case reflect.Int:
```

Comment:
this adds support for letting the user decide how an `int` value is encoded. If your schema defines the field as int64 it will convert to int64. If your schema defines it as int32, it will output as int32. The old behavior was to always choose int32 if if the schema defines it as a target int64 and the downstream processing throws an error on encode because it is expecting an int64.